packages/application-shell/src/components/login/login.spec.js

```diff
@@ -368,48 +360,31 @@ describe('requestAccessToken', () => {
   });
 });
 
-describe('ForgotPassword', () => {
-  describe('when login is not loading', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps();
-      wrapper = shallow(<Login {...props} />);
-    });
-    describe('when clicking "forgot password"', () => {
-      beforeEach(() => {
-        forgotPasswordPreventDefaultMock.mockClear();
-        const clickEvent = { preventDefault: forgotPasswordPreventDefaultMock };
-        wrapper.find('a').simulate('click', clickEvent);
-      });
-      it('preventDefault should be called', () => {
-        expect(forgotPasswordPreventDefaultMock).toHaveBeenCalledTimes(1);
-      });
-      it('should allow redirect', () => {
-        expect(wrapper.state('shouldRedirectPasswordForgot')).toBe(true);
-      });
-    });
+const delay = ms =>
+  new Promise(resolve => {
+    setTimeout(resolve, ms);
   });
-  describe('when login is loading', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps();
-      wrapper = shallow(<Login {...props} />);
-      wrapper.setState({ loading: true });
-    });
-    describe('when clicking "forgot password"', () => {
-      beforeEach(() => {
-        forgotPasswordPreventDefaultMock.mockClear();
-        const clickEvent = { preventDefault: forgotPasswordPreventDefaultMock };
-        wrapper.find('a').simulate('click', clickEvent);
-      });
-      it('preventDefault should not be called', () => {
-        expect(forgotPasswordPreventDefaultMock).toHaveBeenCalledTimes(0);
-      });
-      it('should not allow redirects', () => {
-        expect(wrapper.state('shouldRedirectPasswordForgot')).toBe(false);
-      });
-    });
+
+/* New tests with react-testing-library */
+describe('reset password flow', () => {
+  it('should open a dialog and redirect to the AC', async () => {
+    const props = createTestProps();
+    const { getByText } = renderWithRedux(<Login {...props} />);
+
+    await waitForElement(() => getByText(/Sign in to your account/));
+
+    fireEvent.click(getByText(/Forgot password/));
+
+    await waitForElement(() =>
+      getByText(
+        /We are redirecting you to the Forgot Password form in 3 seconds/
+      )
+    );
+
+    await delay(4000); // the countdown is 3 secods, we wait a bit more just to be sure
```

Comment:
Ah nice, didn't know about that. Thanks!!

Reply:
Oh, and another idea would be to use Jest's [Timer Mocks](https://jestjs.io/docs/en/timer-mocks.html) to forward the timers and make the tests complete sooner instead of actually waiting 3 seconds.

Reply:
Yeah I think controlling time would be the ultimate solution from my view to have runs be speedy.

Reply:
So I looked at the timer mocks but I don't think we can use it for this case because it simply mocks the callback function of the timer. In the test we want to check what happens after.

I ended up doing it like this: https://github.com/commercetools/merchant-center-application-kit/pull/211/commits/ff28439fac111f264be215bb0b6d4c4cfb7591c6

Reply:
It's now using the fake timers 🏊